plugin/filter.go

```diff
@@ -0,0 +1,310 @@
+package main
+
+import (
+	"bytes"
+	"github.com/corazawaf/coraza/v3/types"
+	"github.com/envoyproxy/envoy/contrib/golang/common/go/api"
+	"net"
+	"net/http"
+	"runtime"
+	"strconv"
+)
+
+type filter struct {
+	callbacks           api.FilterCallbackHandler
+	conf                configuration
+	wafMaps             wafMaps
+	tx                  types.Transaction
+	httpProtocol        string
+	isInterruption      bool
+	processRequestBody  bool
+	processResponseBody bool
+}
+
+func (f *filter) DecodeHeaders(headerMap api.RequestHeaderMap, endStream bool) api.StatusType {
+	f.callbacks.Log(api.Info, BuildLoggerMessage(api.Info).msg("DecodeHeaders"))
+	var host string
+	exist := true
+	host = headerMap.Host()
+	if len(host) == 0 {
+		host, exist = headerMap.Get("host")
```

Comment:
Envoy should already handle "Host" & ":authority". In what situation the host is missing?